### PR TITLE
Refactor DistributionTarget.

### PR DESCRIPTION
**Pull request by jsirois**
Refactor DistributionTarget into Target and two concreate subclasses.
This unifies the interface of a Target to make way for a third Target
implementation powered by `pex3 interpreter inspect` data.

Work towards #1597.

### CODE REVIEW

File: pex/pep_425.py [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+    from pex.third_party import attr
+
+
+def _prepare_tags(tags):
+    # type: (Iterable[Tag]) -> Tuple[Tag, ...]
+    return tags if isinstance(tags, tuple) else tuple(OrderedSet(tags))
+
+
+@attr.s(frozen=True)
+class CompatibilityTags(object):
```
Comment (jsirois) [inline pex/pep_425.py:26]: This is new.

File: pex/targets.py [hunk 2]
```diff
@@ -0,0 +1,183 @@
...
+if TYPE_CHECKING:
+    from typing import Iterator, Optional, Tuple
+
+    import attr  # vendor:skip
+else:
+    from pex.third_party import attr
+
+
+@attr.s(frozen=True)
+class Target(object):
```
Comment (jsirois) [inline pex/targets.py:25]: This used to be `DistributionTarget` and was changed from a 2-state single type to two concrete types `LocalInterpreter` and `AbbreviatedPlatform` deriving from this single base class.

File: pex/targets.py [hunk 3]
```diff
@@ -0,0 +1,183 @@
...
+
+    manylinux = attr.ib()  # type: Optional[str]
+
+    def get_supported_tags(self):
+        # type: () -> CompatibilityTags
+        return self.platform.supported_tags(manylinux=self.manylinux)
+
+
+@attr.s(frozen=True)
+class Targets(object):
```
Comment (jsirois) [inline pex/targets.py:147]: This used to be `DistributionTargets`.

File: tests/test_targets.py [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+from pex.targets import AbbreviatedPlatform, LocalInterpreter, Target, Targets
+
+
+@pytest.fixture
+def current_interpreter():
+    # type: () -> PythonInterpreter
+    return PythonInterpreter.get()
+
+
+def test_interpreter(
```
Comment (jsirois) [inline tests/test_targets.py:20]: These tests were mainly just moved from `tests/test_distribution_targets`. The mutex tests of the 2-state `DistributionTarget` type were obsoleted and killed.

File: pex/pep_425.py [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+    """An ordered set of PEP-425 compatibility tags.
+
+    See: https://www.python.org/dev/peps/pep-0425/#use
+    """
+
+    @classmethod
+    def from_stings(cls, tags):
```
Comment (benjyw) [inline pex/pep_425.py:33]:
```suggestion
    def from_strings(cls, tags):
```

**Review by benjyw — APPROVED** [review]
LGTM mod the typo

File: pex/pep_425.py (hunk 5 above)
Comment (jsirois) [reply pex/pep_425.py:33]: I'll fix this in #1609 since that will need another PR burn for a doc change.

Comment (jsirois) [reply pex/pep_425.py:33]: Fixed over in #1609 here: https://github.com/pantsbuild/pex/pull/1609/commits/89bcd0fcab0d1a425be3fef85264a0b1cbbd49a1

**Pull request merged**